### DISCUSSION TITLE
Can we assign styles to the eventcatcher DOM Node?

### DISCUSSION THREAD

**Discussion by BurningTreeC** [TiddlyWiki/TiddlyWiki5#5888] [Ideas]
Hello there,

it would be very useful, if we could assign inline styles directly to the DOM Node created by the `eventcatcher` widget.

is there something that blocks us from realizing this?

**Comment by pmario** [comment]
Most widgets have a class parameter. So I think this should also be possible with this one

**Reply by BurningTreeC** [reply]
Hi @pmario, I'm referring to situations where inline styles are preferred over CSS declarations by class names

**Comment by Jermolene** [comment]
I think we should consider a PR that updates all the widgets that generate DOM nodes where necessary to offer both a `class` and `style` attribute.

**Reply by BurningTreeC** [reply]
Hi @Jermolene, I'm thinking about creating such a PR. Would that mean that every widget that creates a DOM node should get a `assignDomNodeStyles` class?

**Reply by Jermolene** [reply]
Hi @BurningTreeC a PR would be great. I'm not sure what you mean by the `assignDomNodeStyles` class, that's not something in the source code at the moment. I think all that is required is to duplicate the logic in (eg) button.js that handles the `style` and `class` attributes and add it to any widgets that create DOM nodes that don't already handle that attribute.
